### PR TITLE
MULE-18619: Component models not set for infrastructure types in AST

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/extensions-xml-support/src/test/java/org/mule/runtime/extension/internal/loader/validation/DefaultModelValidatorTestCase.java [hunk 1]
```diff
@@ -220,17 +221,17 @@ private ExtensionModel getExtensionModelFrom(String modulePath, Set<ExtensionMod
   private Set<ExtensionModel> getDependencyExtensions() {
     ExtensionModel petstore = loadExtension(PetStoreConnector.class, emptySet());
     ExtensionModel heisenberg = loadExtension(HeisenbergExtension.class, emptySet());
-    return ImmutableSet.<ExtensionModel>builder().add(petstore, heisenberg).build();
+    return ImmutableSet.<ExtensionModel>builder().add(petstore, heisenberg, getExtensionModel()).build();
   }
 
-  private ExtensionModel loadExtension(Class extension, Set<ExtensionModel> deps) {
+  private ExtensionModel loadExtension(Class extension, Set<ExtensionModel> extensions) {
```
Comment (marianogonzalez) [inline modules/extensions-xml-support/src/test/java/org/mule/runtime/extension/internal/loader/validation/DefaultModelValidatorTestCase.java:227]: should this be called depedencyExtensions or something like that? The intent of the parameter is now lost

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java [hunk 2]
```diff
@@ -236,38 +238,65 @@ private static void handleNestedParameters(ComponentModel componentModel, Stream
...
-              } else {
-                enrichComponentModels(componentModel, nestedComponents,
-                                      of(extensionModelHelper.resolveDslElementModel(paramModel,
-                                                                                     componentModel
-                                                                                         .getIdentifier())),
-                                      paramModel, extensionModelHelper);
-              }
-
-            }));
+    model.getAllParameterModels()
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java:241]: seems like nothing change here other than formatting? Is taht the case? Can it be reverted then?

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java [hunk 3]
```diff
@@ -309,26 +334,46 @@ private static void enrichComponentModels(ComponentModel componentModel,
...
+        });
+      }
+
+      @Override
+      public void visitUnion(UnionType unionType) {
+        unionType.getTypes()
+            .forEach(type -> getTypeId(type)
+                .flatMap(paramDsl::getChild)
+                .flatMap(childDsl -> getIdentifier(childDsl))
+                .flatMap(childComponentId -> ofNullable(nestedWrappedComponents.get(childComponentId)))
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java:370]: the flatMaps in a row?? ufff...

Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java:370]: The revenge of the `Optional`s

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java (hunk 2 above)
Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/type/ApplicationModelTypeUtils.java:241]: no, i turned this around, before it iterated the children and matched those against the parameter models, now it iterates the parameter models and looks it up in the dsl.

**Pull request merged**